File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/documents/DocumentService.java
package io.zenwave360.example.clinicaltool.modules.documents;

import io.zenwave360.example.clinicaltool.modules.documents.domain.*;
import io.zenwave360.example.clinicaltool.modules.documents.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 * Inbound Service Port for managing [DocumentInfo].
 */
@org.springframework.modulith.NamedInterface("DocumentService")
public interface DocumentService {

    /**
     *
     *
     */
    public List<DocumentInfo> listDocumentInfo(List<Long> documentIds);
    /**
     *
     *
     */
    public DocumentInfo downloadDocument(Long id);
    /**
     *
     *
     */
    public DocumentInfo uploadDocument(DocumentInfo input);
    /**
     *
     *
     */
    public void deleteDocumentInfo(Long id);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/documents/DocumentService.kt
package io.zenwave360.example.clinicaltool.modules.documents

import io.zenwave360.example.clinicaltool.modules.documents.domain.*
import io.zenwave360.example.clinicaltool.modules.documents.dtos.*
import java.math.*
import java.time.*

/** Inbound Service Port for managing [DocumentInfo]. */
@org.springframework.modulith.NamedInterface("DocumentService")
interface DocumentService {

    /**  */
    fun listDocumentInfo(documentIds: List<Long>?): List<DocumentInfo>

    /**  */
    fun downloadDocument(id: Long): DocumentInfo

    /**  */
    fun uploadDocument(input: DocumentInfo): DocumentInfo

    /**  */
    fun deleteDocumentInfo(id: Long): Unit
}
